src/lua/luadebug.lua

```diff
@@ -0,0 +1,771 @@
+--[[
```

Comment:
@ligurio 
1. I've renamed `cdebugger.lua` to `luadebug.lua` to make it more Lua-specific, and less confusing;
2. I've not yet put it to `third_party` as it would be looking very alone in a separate directory (and there is no yet prior art with similar cases). We need to discuss it further, may be `third_party/lua` would be a good start for later collection;
3. I've reordered refactoring patches to make them acting in reasonable order;
4. And have added unittest for a new LuaC global symbol, and a new luatest for interactive debugging session. It's rather brief, and not cover much commands. (And due to missing breakpoint support, which is yet to come in next pull requests, we still miss an important functionality enabled).

Already developed, but not included features:
- convenient debugger wrapper (tdbg) and/or option `-d`. Need to be discussed and selected;
- proper readline support which would be compatible with Tarantool console. It's hanging if activated from console;
- breakpoints.

Reply:
After discussion with @igormunkin I've moved `luadebug.lua` to `third_party/lua/luadebug.lua`. The plan is to gradually refactor out relevant code from there, moving to separate modules which will be used both by console debugger `luadebug.lua` and by forthcoming DAP debugging layer. Eventually the code left there will become nil, or negligible, and we will get rid of it.